### PR TITLE
Add dynamic indexing of elements for raw demo page

### PR DESCRIPTION
**Pull request by castastrophe**
This adds a script to dynamically list all components on the demo page when you run `npm run start`

### CODE REVIEW

**Comment by mwcz** [comment]
Lol, a commit message of "--append" was not exactly what I meant to do...

File: scripts/doc-listing-update.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  // Output the updated index file
+  fs.writeFile(index, content, function(err) {
+    if (err) throw err;
+  });
+});
+
+// From the wrapper.html, get the ID element #demos and replace with the above markup
+const outFile = path.join(__dirname, "../doc/index.html");
+
+// fs.writeFileSync(outFile);
```
[the remark this answers is not in the document]
Comment (mwcz) [reply scripts/doc-listing-update.js]: @kylebuch8 `writeFile` is called above, within `readFile`'s callback.

**Comment by mwcz** [comment]
I made a small set of changes, @castastrophe please look over them and see what you think.  I can revert any that you don't like.

First change was to rename `doc-listing-update.js` to `doc-listing-inject.js` so it has a similar name to `test-suite-inject.js`, as they do very similar things.  I also renamed the `index` npm script to `doc-listing-inject` to make it clear they're connected.

Second change was creating a `post-new` npm script which gets run after `npm run new`.  There were already a couple of scripts that needed to be run after elements are created (test-suite-inject and bootstrap), so I thought it would be good to consolidate them into a single place.  I added the new script to `post-new` and removed it from `start`.  I think that covers 

Third change was to remove the commented code that Kyle mentioned.

**Comment by castastrophe** [comment]
@mwcz Lettuce, Gruyère, Tomato, and Mutton

**Pull request merged**